vllm/model_executor/sampling_metadata.py

```diff
@@ -223,8 +224,7 @@ def _prepare_seq_groups(
 
         if seq_group_metadata.is_prompt:
             if sampling_params.seed is not None:
-                generator = torch.Generator(device=device).manual_seed(
-                    sampling_params.seed)
+                generator = np.random.default_rng(seed=sampling_params.seed)
```

Comment:
Switched to NumPy generator because the random number generation at CPU can overlap with other GPU work, reducing the overall latency.